### PR TITLE
docs

### PR DESCRIPTION
**Pull request by kkoch986**
- **WIP: start outlining the documentation**
- **fixup! WIP: start outlining the documentation**

### CODE REVIEW

File: src/api/listing/taxonomy.proto [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  // for top-level categories
+  string parent_category_id = 2;
+  // A user-friendly name for the category
+  string display_name = 3;
+
+  google.protobuf.Timestamp updated_at = 4;
+  google.protobuf.Timestamp created_at = 5;
+}
+
+// AttributeSpec is the definition of an attribute on a channel.
```
Comment (mjthomas588) [inline src/api/listing/taxonomy.proto:28]: I think it's a little clearer to say that it's the definition of an Attribute within the given Category for a Channel.

File: src/api/listing/taxonomy.proto [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+// The spec also provides information about how to validate Attributes and how
+// to derive their value from the catalog system.
+// The link between an AttributeSpec and a Category indicates the
```
Comment (mjthomas588) [inline src/api/listing/taxonomy.proto:31]: I think we should remove this line. makes me think you can take an existing attribute spec and associate it with a category

File: src/api/listing/taxonomy.proto [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  string parent_category_id = 2;
+  // A user-friendly name for the category
+  string display_name = 3;
+
+  google.protobuf.Timestamp updated_at = 4;
+  google.protobuf.Timestamp created_at = 5;
+}
+
+// AttributeSpec is the definition of an attribute on a channel.
+// The spec also provides information about how to validate Attributes and how
```
Comment (mjthomas588) [inline src/api/listing/taxonomy.proto:29]: I don't think they should be providing or accessing information about how to derive an attributes value from the catalog system. maybe we leave that part out.

File: src/api/listing/taxonomy.proto [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  Classification classification = 8;
+
+  google.protobuf.Timestamp updated_at = 9;
+  google.protobuf.Timestamp created_at = 10;
+}
+
+// ProductType are detailed classifications dictated by the channel.
+// They should be at least as specific as the Category.
+// Generally they should be more specific than the Category.
+// ProductTypes are only used in the process of deriving ListingQuality.
```
Comment (mjthomas588) [inline src/api/listing/taxonomy.proto:95]: maybe it's a little clearer to say "Product Types are used to specify which attributes are required and recommended for a given Listing"

File: src/api/listing/taxonomy_service.proto [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+  }
+
+  // UpdateCategory will update the Category with the given ID
+  rpc UpdateCategory(UpdateCategoryRequest) returns (UpdateCategoryResponse) {
+    option (google.api.http) = {
+      put: "/v2/storefront/taxonomy/category",
+      body: "*"
+    };
+  }
+
```
Comment (mjthomas588) [inline src/api/listing/taxonomy_service.proto:44]: there's no `DeleteCategory`

**Review by mjthomas588 — APPROVED** [review]

File: src/api/listing/taxonomy.proto (hunk 2 above)
Comment (kkoch986) [reply src/api/listing/taxonomy.proto:31]: yea that makes sense, ill remove it

File: src/api/listing/taxonomy.proto (hunk 3 above)
Comment (kkoch986) [reply src/api/listing/taxonomy.proto:29]: oops yea, i think that was just copied over from the internal one

File: src/api/listing/taxonomy_service.proto (hunk 5 above)
Comment (kkoch986) [reply src/api/listing/taxonomy_service.proto:44]: hm we dont have one on the internal service yet either, maybe we just build it once its required.

File: src/api/listing/taxonomy.proto (hunk 4 above)
Comment (kkoch986) [reply src/api/listing/taxonomy.proto:95]: like it, updated

File: guide/taxonomy-ingestion.md [hunk 6]
```diff
@@ -76,65 +67,24 @@ with a body like:
...
-
-## Product Types
+### Listing Level
 
-At this point, we have established the bear minimum for generating a listing
-from Zentail. The next 2 sections outline tools you can use to provide greater
-visibility into what makes a great listing on your channel. `ProductType`s and 
-`QualityAttributeSpec` drive Listing Quality workflows in Zentail that help sellers
-understand requirements and recommendations for each Listing.
+The `level` property can be used to enforce that certain values must share a common value
```
Comment (dsugarman) [inline guide/taxonomy-ingestion.md:80]: this isn't totally clear to me, it should say to use `LEVEL_LISTING` when you want to ensure that an attribute needs to have the same value for all variants.  i think the next callout saying it's fine to just put LEVEL_VARIANT for everything looks good

File: guide/taxonomy-ingestion.md [hunk 7]
```diff
@@ -13,54 +13,45 @@ role in the integration is.
...
 
 ```json
 {
-  "category": {
-    "channelCategoryId": "<an identifier for your channel, must be unique>",
-    "parentCategoryId": "<the ID of the parent category, can be omitted or set to an empty string to indicate root level>",
-    "displayName": "<a user friendly display name>"
+  "productType": {
+    "productTypeId": "<the productType ID to associate the product type with>",
+    "channelProductTypeId": "<your id for this product type>"
```
Comment (dsugarman) [inline guide/taxonomy-ingestion.md:34]: what's the difference between this and productTypeId?

File: guide/taxonomy-ingestion.md [hunk 8]
```diff
@@ -13,54 +13,45 @@ role in the integration is.
...
+POST /v2/storefront/taxonomy/attribute/{productTypeId}
 ```
 
 with a body like:
 
 ```jsonc
 {
-  "attributeSpec": {
-    "attributeSpecId": "<a unique identifier for this attribute, must only be unique within the category ID>",
+  "attribute": {
```
Comment (dsugarman) [inline guide/taxonomy-ingestion.md:53]: where do we specify "productTypeId"?

File: guide/taxonomy-ingestion.md (hunk 7 above)
Comment (kkoch986) [reply guide/taxonomy-ingestion.md:34]: that shouldnt be there, just fixed it to be:

```

```json
{
  "productType": {
    "productTypeId": "<a unique identifier for this product type>",
    "displayName": "< a user-friendly name for this product type>"
  }
}
```

File: guide/taxonomy-ingestion.md (hunk 8 above)
Comment (kkoch986) [reply guide/taxonomy-ingestion.md:53]:
```
POST /v2/storefront/taxonomy/attribute/{productTypeId}
```

its in the request url

**Pull request merged**